app/database/group_group_store.go

```diff
@@ -22,3 +27,53 @@ func (s *GroupGroupStore) WhereUserIsMember(user *User) *DB {
 func (s *GroupGroupStore) createNewAncestors() {
 	s.DataStore.createNewAncestors("groups", "Group")
 }
+
+var ErrRelationAlreadyExists = errors.New("the relation already exists")
+var ErrRelationCycle = errors.New("a group cannot become an ancestor of itself")
+
+const groupsRelationsLockTimeout = 3 * time.Second
+
+func (s *GroupGroupStore) CreateRelation(parentGroupID, childGroupID int64) (err error) {
+	s.mustBeInTransaction()
+	defer recoverPanics(&err)
+
+	mustNotBeError(s.WithNamedLock(s.tableName, groupsRelationsLockTimeout, func(store *DataStore) (err error) {
+		var rows []interface{}
+		mustNotBeError(s.WithWriteLock().
+			Select("ID").
+			Where("idGroupChild = ? AND idGroupParent = ?", childGroupID, parentGroupID).
```

Comment:
If it exists but is "removed" I supposed it can be recreated.

Reply:
from now it will always recreate a relation, no matter which type it has